test/construction_and_conversion.jl

```diff
@@ -8,6 +8,9 @@ global reps = 10
     @test DNAKmer(DNA_G, DNA_C, DNA_T) == Kmer("GCT")
     @test RNAKmer(RNA_G, RNA_U, RNA_C, RNA_U) == Kmer("GUCU")
     
+    # creation from iterator
+    @test Kmers.kmertype(Kmer{DNAAlphabet{2},31})((i for i in rand(ACGT, 31))) isa Kmers.kmertype(Kmer{DNAAlphabet{2},31})
```

Comment:
Specifically tests the example in https://github.com/BioJulia/BioSequences.jl/issues/270